server/s3_object_action_base.cc

```diff
@@ -106,11 +107,95 @@ void S3ObjectAction::fetch_object_info() {
 }
 
 void S3ObjectAction::fetch_object_info_success() {
-  request->set_object_size(object_metadata->get_content_length());
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
   next();
 }
 
-void S3ObjectAction::load_metadata() { fetch_bucket_info(); }
+// For certain APIs like CopyObject, fetch additional (source)
+// bucket metadata/information
+void S3ObjectAction::fetch_additional_bucket_info() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  s3_log(S3_LOG_DEBUG, request_id, "Fetch metadata of bucket: %s\n",
+         additional_bucket_name.c_str());
+
+  additional_bucket_metadata =
+      bucket_metadata_factory->create_bucket_metadata_obj(
+          request, additional_bucket_name);
+
+  additional_bucket_metadata->load(
+      std::bind(&S3ObjectAction::fetch_additional_bucket_info_success, this),
+      std::bind(&S3ObjectAction::fetch_additional_bucket_info_failed, this));
+
+  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
```

Comment:
lets log bucket name instead of fuction exit

Reply:
bucket name i being logged at start of this function